packages/editor/src/editor-styles.scss

```diff
@@ -1,19 +1,109 @@
+/**
+ * Editor Normalization Styles
+ *
+ * These are only output in the editor, but styles here are prefixed .editor-styles-wrapper and affect the theming
+ * of the editor by themes.
+ * Why do these exist? Why not rely on browser defaults?
+ * These styles are necessary so long as CSS can bleed from the wp-admin into the editing canvas itself.
+ * Let's continue working to refactor these away, whether through Shadow DOM or better scoping of upstream styles.
+ */
+
 body {
 	font-family: $editor-font;
 	font-size: $editor-font-size;
 	line-height: $editor-line-height;
 	color: $dark-gray-900;
 }
 
+/* Headings */
+// These follow a Major Third type scale
+// https://type-scale.com/?size=16&scale=1.250&text=A%20Visual%20Type%20Scale&font=Noto%20Serif&fontweight=600
+h1 {
+	font-size: 2.44em;
+}
+
+h2 {
+	font-size: 1.95em;
+}
+
+h3 {
+	font-size: 1.56em;
+}
+
+h4 {
+	font-size: 1.25em;
+}
+
+h5 {
+	font-size: 1em;
+}
+
+h6 {
+	font-size: 0.8em;
+}
+
+// Adjust line spacing for larger headings.
+h1,
+h2,
+h3 {
+	line-height: 1.4;
+}
+
+h4 {
+	line-height: 1.5;
+}
+
+// Default margins.
+h1 {
+	margin-top: 0.67em;
+	margin-bottom: 0.67em;
+}
+
+h2 {
```

Comment:
Are these normalization styles supposed to be inlined and prefixed with `.editor-styles-wrapper`?
I'm seeing them all override my styles in the editor.

Reply:
As usual: thank you for testing! I know you're busy, but I much appreciate your productive feedback.

In this branch, the rules are now `.editor-styles-wrapper h1 { ... }` for h1 to h6, and they need some kind of prefix to override the wp-admin rules. 

In master, the rules are `.wp-block-heading h1 { ... }`, which is enough to override the wp-admin rules, but those rules also seem _more_ specific to me, don't they?

> Are these normalization styles supposed to be inlined and prefixed with .editor-styles-wrapper?

Yes, this is already the case in master, and is an effort to make it easier for themes to write simple CSS and not have to worry about styling all of wp-admin in the process, and for the styles to receive higher specificity in the editor. This also allows your editor styles to apply to variations previews which you'll see if you open the block switcher for, for example a quote — you'll see the various variations of the quote in your editor style.

Any rule your theme provides in the editor style you load using `add_editor_style` will also be [rewritten on the fly](https://wordpress.org/gutenberg/handbook/designers-developers/developers/themes/theme-support/#editor-styles) to prefix `.editor-styles-wrapper` to all your rules. This is also why they are inlined. Which means if you write:

```
/* My custom editor style */
h2 {
margin: 2em 0;
color: cyan;
}
```

This will be rewritten to the following, before it is loaded into the editing canvas:

```
/* My custom editor style */
.editor-styles-wrapper h2 {
margin: 2em 0;
color: cyan;
}
```

So if you are loading your CSS using add_editor_style, your rules should work properly in the canvas, and should easily override the normalization styles you refer to, because they are output later in the DOM.

Reply:
Oh, I see now. 
I'm using `.editor-styles-wrapper h2 {` in my themes but I'm not inlining it. 
So the `inline` `.editor-styles-wrapper h2 {` wins. 
My reasons for not using `editor_style` theme_support are out of scope here but in short, it breaks `:root {` selectors and it allows my element styles to mutate core editor components. I actually wrote a [PostCSS plugin](https://github.com/m-e-h/postcss-editor-styles) to handle this for me. I can increase my specificity there. Is a shame that I need to but not a problem. :wink: 

I did a quick visual check of TwentyNineteen last night. Not seeing much change if any. TwentyNineteen probably isn't the best example for this though since it's margins were already pretty close to the editor margin size.

![editor-before_after](https://user-images.githubusercontent.com/3585901/55564931-ebbc2480-56c6-11e9-8d67-271c8869c373.png)

Hard to tell with some of the other differences between front and editor. Line-heights look different and I'm seeing a difference in some of the fonts (the apostrophe in You're). 

Maybe a little more accurate margins with these changes?